### PR TITLE
Organize Dependencies.scala

### PR DESCRIPTION
**Pull request by jmthibault79**
No JIRA, just one of my "refactor for funsies" PRs

- [ ] **Submitter**: Include the JIRA issue number in the PR description
- [ ] **Submitter**: Check that the **Product Owner** has signed off on any user-facing changes
- [ ] **Submitter**: Make sure Swagger is updated if API changes
  - [ ] **...and Orchestration's Swagger too!**
- [ ] **Submitter**: If updating admin endpoints, also update [firecloud-admin-cli](https://github.com/broadinstitute/firecloud-admin-cli)
- [ ] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [ ] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * Add notes on what you've tested
- [ ] **Submitter**: Update RC_XXX release ticket with any config or environment changes necessary
- [ ] **Submitter**: Database checks:
  * If PR includes new or changed db queries, include the explain plans in the description
  * Make sure liquibase is updated if appropriate
  * If doing a migration, take a backup of the
  [dev](https://console.cloud.google.com/sql/instances/terraform-qfarbdq3lrexxck5htofjs5z6m/backups?project=broad-dsde-dev&organizationId=548622027621)
  and
  [alpha](https://console.cloud.google.com/sql/instances/terraform-r4caezzc35c4tb7pgdhwkmme4y/backups?project=broad-dsde-alpha&organizationId=548622027621)
  DBs in Google Cloud Console
- [ ] **Submitter**: Update FISMA documentation if changes to:
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
- [ ] Tell your tech lead (TL) that the PR exists if they want to look at it
- [ ] Anoint a lead reviewer (LR). **Assign PR to LR**
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
- [ ] **TL** sign off
- [ ] **LR** sign off
- [ ] **Assign to submitter** to finalize
- [ ] **Submitter**: Verify all tests go green, including CI tests
- [ ] **Submitter**: Squash commits and merge to develop
- [ ] **Submitter**: Delete branch after merge
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Verify swagger UI on dev environment still works after deployment
- [ ] **Submitter**: Inform other teams of any API changes via hipchat and/or email
- [ ] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

File: project/Dependencies.scala [hunk 1]
```diff
@@ -5,93 +5,109 @@ object Dependencies {
   val sprayV = "1.3.3"
   val slickV = "3.1.1"
 
+  val googleV = "1.22.0"
+  val olderGoogleV = "1.20.0"   // TODO why do we have two google versions?
```
Comment (jmthibault79) [inline project/Dependencies.scala]: So this is interesting.  Haven't investigated this yet.

File: project/Dependencies.scala [hunk 2]
```diff
@@ -5,93 +5,109 @@ object Dependencies {
...
-    "com.typesafe.akka" %% "akka-actor" % akkaV,
-    "com.google.apis" % "google-api-services-cloudbilling" % "v1-rev7-1.22.0"
-      exclude("com.google.guava", "guava-jdk5"),
-    "com.google.apis" % "google-api-services-genomics" % "v1-rev89-1.22.0"
-      exclude("com.google.guava", "guava-jdk5"),
-    "com.google.apis" % "google-api-services-storage" % "v1-rev35-1.20.0"
-      exclude("com.google.guava", "guava-jdk5"),
-    "com.google.apis" % "google-api-services-cloudresourcemanager" % "v1-rev7-1.22.0"
-      exclude("com.google.guava", "guava-jdk5"),
-    "com.google.apis" % "google-api-services-cloudbilling" % "v1-rev7-1.22.0"
```
Comment (jmthibault79) [inline project/Dependencies.scala:20]: So nice we used to include it twice

File: project/Dependencies.scala [hunk 3]
```diff
@@ -5,93 +5,109 @@ object Dependencies {
...
 
-  val rawlsCoreDependencies = modelDependencies ++ Seq(
+  val rawlsCoreDependencies = modelDependencies ++ googleDependencies ++ Seq(
     "com.typesafe" % "config" % "1.3.0",
     "com.gettyimages" %% "spray-swagger" % "0.5.0"
       exclude("com.typesafe.scala-logging", "scala-logging-slf4j_2.11")
       exclude("com.typesafe.scala-logging", "scala-logging-api_2.11")
       exclude("com.google.guava", "guava"),
     "com.getsentry.raven" % "raven-logback" % "7.8.6",
-    "com.typesafe.akka" %% "akka-actor" % akkaV,
```
Comment (jmthibault79) [inline project/Dependencies.scala]: included in googleDependencies.  Could call it out explicitly here

File: project/Dependencies.scala [hunk 4]
```diff
@@ -5,93 +5,109 @@ object Dependencies {
...
+    slickHikariCP,
+    sprayCan,
+    sprayRouting,
+    sprayClient,
     "org.webjars" % "swagger-ui" % "2.1.1",
     "org.apache.commons" % "commons-jexl" % "2.1.1",
     "org.broadinstitute" %% "wdl4s" % "0.11" exclude("org.scalaz", "scalaz-core_2.11") exclude("io.spray", "spray-json_2.11"),
     "org.scalaz" % "scalaz-core_2.11" % "7.1.3",
     "org.broadinstitute.dsde.vault" %% "vault-common" % "0.1-15-bf74315",
-    "com.google.apis" % "google-api-services-cloudbilling" % "v1-rev7-1.22.0"
```
Comment (jmthibault79) [inline project/Dependencies.scala]: These are all in `googleDependencies`.  `rawlsCoreDependencies` included all of `googleDependencies` already, so I was able to just add the whole thing above.

File: project/Dependencies.scala [hunk 5]
```diff
@@ -5,93 +5,109 @@ object Dependencies {
...
+    jodaTime,
+    jodaConvert,
+    scalaLogging,
+    googleApiClient,
+    scalatest
   )
 
-  val rawlsCoreDependencies = modelDependencies ++ Seq(
+  val rawlsCoreDependencies = modelDependencies ++ googleDependencies ++ Seq(
     "com.typesafe" % "config" % "1.3.0",
```
Comment (jmthibault79) [inline project/Dependencies.scala]: Could make all these `val`s too, for consistency.

**Comment by helgridly** [comment]
holy shit this is glorious

File: project/Dependencies.scala (hunk 5 above)
Comment (jmthibault79) [reply project/Dependencies.scala]: yeah I'm gonna do that

Comment (helgridly) [reply project/Dependencies.scala]: joel PRing so hard he's talking to himself in comments 👍

File: project/Dependencies.scala (hunk 1 above)
Comment (helgridly) [reply project/Dependencies.scala]: does git blame give you any interesting info here?

File: project/Dependencies.scala [hunk 6]
```diff
@@ -5,93 +5,126 @@ object Dependencies {
...
+    ravenLogback,
+    slick,
+    slickHikariCP,
+    sprayCan,
+    sprayRouting,
+    sprayClient,
+    swaggerUI,
+    commonsJEXL,
+    wdl4s,
+    scalaz,
```
Comment (helgridly) [inline project/Dependencies.scala]: This popped out at me -- I don't think it's used anywhere. Can you check and remove it if not?
(I also can't find a reference to cats, which means it's pulled in transitively by something else. Hmm.)

**Comment by helgridly** [comment]
So I guess the baseline assumption here is that we want to use the same versions of any given library across our modules (e.g. spray-http in both google and model). I don't see why this wouldn't be true but it's worth checking. @dvoet ?

File: project/Dependencies.scala (hunk 6 above)
Comment (rtitle) [reply project/Dependencies.scala]: `sbt dependencyTree`

Looks like `org.broadinstitute:wdl4s` is pulling it in:
```
+-org.broadinstitute:wdl4s_2.11:0.11
| +-org.typelevel:cats_2.11:0.9.0
```
I agree about removing the scalaz dependency and adding an explicit cats dependency in Rawls if we're actually relying on it. (Maybe also exclude it from wdl4s).

File: project/Dependencies.scala [hunk 7]
```diff
@@ -5,93 +5,126 @@ object Dependencies {
...
+  val googleCloudBilling: ModuleID =          excludeGuavaJDK5("com.google.apis"        % "google-api-services-cloudbilling"          % ("v1-rev7-" + googleV))
+  val googleGenomics: ModuleID =              excludeGuavaJDK5("com.google.apis"        % "google-api-services-genomics"              % ("v1-rev89-" + googleV))
+  val googleStorage: ModuleID =               excludeGuavaJDK5("com.google.apis"        % "google-api-services-storage"               % ("v1-rev35-" + olderGoogleV))
+  val googleCloudResourceManager: ModuleID =  excludeGuavaJDK5("com.google.apis"        % "google-api-services-cloudresourcemanager"  % ("v1-rev7-" + googleV))
+
+  val googleCompute: ModuleID =           "com.google.apis"   % "google-api-services-compute"           % ("v1-rev72-" + olderGoogleV)
+  val googleAdminDirectory: ModuleID =    "com.google.apis"   % "google-api-services-admin-directory"   % ("directory_v1-rev53-" + olderGoogleV)
+  val googlePlus: ModuleID =              "com.google.apis"   % "google-api-services-plus"              % ("v1-rev381-" + olderGoogleV)
```
Comment (rtitle) [inline project/Dependencies.scala:35]: Why do Google cloud libraries have such crazy versions? `v1-rev381-1.20.0` WTF?

`aws-java-sdk` versions are nice and clean :) 
https://mvnrepository.com/artifact/com.amazonaws/aws-java-sdk

**Comment by rtitle** [comment]
Looks great! 👍 if tests pass.

File: project/Dependencies.scala (hunk 1 above)
Comment (jmthibault79) [reply project/Dependencies.scala]: Not specifically.  Looks like the newer ones were just added more recently.  We could add a tech debt story to update/align these.

**Pull request merged**